### PR TITLE
Make VCS URI annotation configurable

### LINKED ISSUE #38055: Make annotation app.quarkus.io/vcs-uri optional in Kubernetes extension
### Description

The Kubernetes extension adds the annotation `app.quarkus.io/vcs-uri` to the generated deployment.

However, in our case it is integrated in our GitLab CI pipelines, so when the repository is cloned the GL user and token are specified in the URL and as a consequence the token is written as part of the annotation:
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    app.quarkus.io/vcs-uri: https://gitlab-ci-token:<token>@gitlab.onprem.com/project.git
...
```

It should be possible to opt-out for this annotation.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by xstefank**
- Closes: #38055

### CODE REVIEW

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java [hunk 1]
```diff
@@ -1191,4 +1191,11 @@ private static List<PolicyRule> toPolicyRulesList(Map<String, PolicyRuleConfig>
                         .build())
                 .collect(Collectors.toList());
     }
+
+    private static String parseVCSUri(VCSUriConfig config, ScmInfo scm) {
+        if (config.enabled) {
+            return config.override.orElseGet(() -> scm != null ? Git.sanitizeRemoteUrl(scm.getRemote().get("origin")) : null);
```
Comment (jcechace) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java:1197]: Mirroring my comment from the original issue. 

Is it reasonable to get ScmInfo directly from SCM configuration compared to SCM configuration of project's build file (e.g. `pom.xml`)? I really wouldn't expect application framework to go digging into my git configuration.

**Comment by jcechace** [comment]
Left a food for thought comment. Regardless thank you for providing an override option!

**Review by geoand — APPROVED** [review]

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java (hunk 1 above)
Comment (iocanel) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java:1197]: SCM configuration may or may not be configured, in contrast to `.git`.
Do you see any reason why the extension should not go digging to the git configuration?

**Review by iocanel — APPROVED** [review]
LGTM

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java (hunk 1 above)
Comment (geoand) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java:1197]: Yeah, I really don't understand the argument of not using `.git`...

**Pull request merged**